File: palindrome/js/palindrome.js
exports.palindrome = function(word) {
    // Create a regex to test each charcter. 
    // The character should only match letters and numbers.
    let edgeRegex = /\w/;
    // Split the word into an array
    let wordArr = String(word).split('');
    // Filter characters to eliminate any that don't pass regex
    wordArr = wordArr.filter(x => edgeRegex.test(x));
    // Create array to store reversed characters
    let reverseWordArr = [];
    // For loop adds characters to the front of array
    for (char of wordArr) {
        reverseWordArr.unshift(char);
    }
    // compare the two arrays
    if (wordArr.join('').toLowerCase() === reverseWordArr.join('').toLowerCase()) {
        return true;
    }

    return false;
};

File: palindrome/python/palindrome.py
import re

def palindrome(word):
    # Create a regex to test each character.
    edgeRegex = r'\w'
    # Ensure the word is a string
    word = str(word)
    # Initialize a string variable to hold all passing chars
    newWord = ''
    # For loop to add all passing chars to newWord 
    for char in word:
        match = re.match(edgeRegex, char)
        if match:
            newWord += char
    # Variable for reversed string
    reversedword = newWord[::-1]
    # Compare the newWord and reversedWord
    if newWord.lower() == reversedword.lower():
        return True
    
    return False
